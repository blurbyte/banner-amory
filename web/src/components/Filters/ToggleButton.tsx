// animated button at the bottom

import * as React from 'react';
import styled from 'styled-components';

import { SimpleArrow } from '../Icons';
import Button from './Button';

const Wrapper = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  position: absolute;
  left: 0;
`;

type ToggleButtonProps = {
  onClick: () => void;
  isExpanded?: boolean;
};

class ToggleButton extends React.Component<ToggleButtonProps> {
  render() {
    return (
      <Wrapper>
        <Button type="button" aria-label="Toggle filters" {...this.props}>
          <SimpleArrow />
        </Button>
      </Wrapper>
    );
  }
}

export default ToggleButton;
