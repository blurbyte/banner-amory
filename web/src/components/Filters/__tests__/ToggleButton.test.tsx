import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import ToggleButton from '../ToggleButton';

test('renders correctly', () => {
  const props = {
    onClick: jest.fn(),
    isExpanded: true
  };

  const component = renderer.create(<ToggleButton {...props} />);
  expect(component.toJSON()).toMatchSnapshot();
});
