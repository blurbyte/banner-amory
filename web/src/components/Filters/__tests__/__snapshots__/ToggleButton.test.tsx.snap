// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  color: #ffffff;
  padding: 0;
  padding-bottom: 0.8rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  width: 10rem;
  height: 2rem;
  border: none;
  cursor: pointer;
  outline: none;
  -webkit-transform: rotateX(180deg);
  -ms-transform: rotateX(180deg);
  transform: rotateX(180deg);
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: absolute;
  left: 0;
}

<div
  className="c0"
>
  <button
    aria-label="Toggle filters"
    className="c1"
    onClick={[MockFunction]}
    type="button"
  >
    <svg
      height="12"
      viewBox="0 0 24 12"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24L12 12 0 0z"
        fill="currentColor"
      />
      <path
        d="M6 0h12l-6 6-6-6z"
        fill="#3d26da"
      />
    </svg>
  </button>
</div>
`;
